atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManagerImpl.java

```diff
@@ -114,6 +115,34 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
         return numTimesFailed > NUM_RETRIES;
     }
 
+    public <C extends PreCommitCondition, E extends Exception> ServiceWriteTransaction getWriteTransaction(C condition)
+            throws E {
+        checkOpen();
+        try {
+            RawTransaction tx = setupRunTaskWithConditionThrowOnConflict(condition);
+            return new ServiceWriteTransaction(tx.delegate()) {
+                @Override
+                public void commit() {
+                    tx.commit();
+                    finishRunTaskWithLockThrowOnConflict(tx, transaction -> null);
```

Comment:
`finishRunTaskWithLocksThrowOnConflict` unlocks the immutableTs and adds stuff to the cleaner if the transaction is a `HARD_DELETE`, so figured we should probably leave them for correctness.

Reply:
Do we want to trust that this method can't throw? Could use try/finally instead.